service/history/api/recordactivitytaskstarted/api.go

```diff
@@ -232,3 +250,29 @@ func recordActivityTaskStarted(
 
 	return response, false, nil
 }
+
+// TODO: move this method to a better place
+// TODO: cache this result (especially if the answer is true)
+func getTaskQueueCurrentDeployment(
+	ctx context.Context,
+	namespaceID string,
+	taskQueueName string,
+	taskQueueType enumspb.TaskQueueType,
+	matchingClient matchingservice.MatchingServiceClient,
+) (*deploymentpb.Deployment, error) {
+	resp, err := matchingClient.GetTaskQueueUserData(ctx,
```

Comment:
If we've been saying the deployment entity wf is the source of truth, shouldn't we check the memo on the entity wf (though a deployment client)?

Checking the user data on the root seems like it might be prone to races.. even after we change the entity wf to wait for user data propagation, the root will still be updated before the propagation finishes.